openwisp_controller/config/tests/test_vpn.py

```diff
@@ -1655,27 +1666,6 @@ def _reset_mocks():
             self.assertEqual(mock_requests.post.call_count, 1)
         _reset_mocks()
 
-        with self.subTest(
-            (
-                'Test zt no identity generation when same zt vpn server '
-                'vpn client template is already applied to the device'
-            )
-        ):
-            device.config.templates.add(zt_1_copy)
-            # No new zt vpn client object
-            self.assertEqual(vpnclient_qs.count(), 1)
-            # Make sure only previously applied 'zt_vpn1' vpn client object is exist
-            self.assertEqual(vpnclient_qs.first().vpn, zt_vpn1)
-            self.assertEqual(IpAddress.objects.count(), 4)
-            # Make sure subprocess is not called for identity generation
-            self.assertEqual(mock_subprocess.run.call_count, 0)
-            # No configuration changed
-            self.assertEqual(mock_info.call_count, 0)
-            self.assertEqual(mock_warn.call_count, 0)
-            self.assertEqual(mock_error.call_count, 0)
-            self.assertEqual(mock_requests.post.call_count, 0)
-        _reset_mocks()
-
```

Comment:
This was verifying that adding a duplicate template for the sample VPN server will not generate a new identity (VpnClient).

With the changes in current PR,  a combination of `vpn`, `config` and `template` is used to determine whether to create VpnClient of the added template.. This test will raise an IntegrityError.